crates/holochain/tests/multi_conductor.rs

```diff
@@ -163,3 +163,192 @@ async fn invalid_cell() -> anyhow::Result<()> {
+                let hash = api
+                    .create_link(CreateLinkInput {
+                        base_address: base_target.0.into(),
+                        target_address: base_target.1.into(),
+                        tag: LinkTag::new(vec![]),
+                    })
+                    .unwrap();
+                Ok(hash)
+            },
+        )
+        .callback("get_links", |api, base: AgentPubKey| {
+            Ok(api
+                .get_links(vec![GetLinksInput {
+                    base_address: base.into(),
+                    tag_prefix: None,
+                }])
+                .unwrap())
+        })
+        .callback("delete_link", |api, hash: HeaderHash| {
+            api.delete_link(hash).unwrap();
+            Ok(())
+        });
+    let mut conductors = SweetConductorBatch::from_config(NUM_CONDUCTORS, config).await;
+
+    let (dna_file1, _) = SweetDnaFile::unique_from_test_wasms(vec![TestWasm::Link])
+        .await
+        .unwrap();
+    let (dna_file2, _) = SweetDnaFile::unique_from_inline_zome("zome1", app)
+        .await
+        .unwrap();
+
+    let apps = conductors.setup_app("app1", &[dna_file1]).await.unwrap();
+    conductors.exchange_peer_info().await;
+
+    let ((alice,), (bobbo,), (carol,)) = apps.into_tuples();
+
+    // Call the "create" zome fn on Alice's app
+    let hash: HeaderHash = conductors[0]
+        .call(&alice.zome(TestWasm::Link), "create_link", ())
+        .await;
+
+    consistency_10s(&[&alice, &bobbo, &carol]).await;
+
+    let links: Links = conductors[0]
+        .call(&alice.zome(TestWasm::Link), "get_links", ())
+        .await;
+    let links = links.into_inner();
+
+    assert_eq!(links.len(), 1);
+
+    let links: Links = conductors[1]
+        .call(&bobbo.zome(TestWasm::Link), "get_links", ())
+        .await;
+    let links = links.into_inner();
+
+    assert_eq!(links.len(), 1);
+
+    let links: Links = conductors[2]
+        .call(&carol.zome(TestWasm::Link), "get_links", ())
+        .await;
+
+    let links = links.into_inner();
+    assert_eq!(links.len(), 1);
+
+    let _: HeaderHash = conductors[1]
+        .call(&bobbo.zome(TestWasm::Link), "delete_link", hash)
+        .await;
+
+    consistency_10s(&[&alice, &bobbo, &carol]).await;
+
+    let links: Links = conductors[0]
+        .call(&alice.zome(TestWasm::Link), "get_links", ())
+        .await;
+    let links = links.into_inner();
+
+    assert_eq!(links.len(), 0);
+
+    let links: Links = conductors[1]
+        .call(&bobbo.zome(TestWasm::Link), "get_links", ())
+        .await;
+    let links = links.into_inner();
+
+    assert_eq!(links.len(), 0);
+
+    let links: Links = conductors[2]
+        .call(&carol.zome(TestWasm::Link), "get_links", ())
+        .await;
+
+    let links = links.into_inner();
+    assert_eq!(links.len(), 0);
+
+    let apps = conductors.setup_app("app", &[dna_file2]).await.unwrap();
+    conductors.exchange_peer_info().await;
```

Comment:
Is everything below this line just the same thing repeated with an inline zome instead of a test wasm? If so could you DRY this up into a loop, with a trace at the beginning saying which DnaFile is being used, to make it clear that it's the same?